sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/assessment-workflow-queue-retry.scheduler.ts

```diff
@@ -0,0 +1,114 @@
+import { InjectQueue, Process, Processor } from "@nestjs/bull";
+import { Job, Queue } from "bull";
+import { BaseScheduler } from "../base-scheduler";
+import { addHours, QueueNames } from "@sims/utilities";
+import { QueueService } from "@sims/services/queue";
+import {
+  StudentAssessmentService,
+  WorkflowEnqueuerService,
+} from "../../../services";
+import {} from "../../models/processors.models";
+import { SystemUsersService } from "@sims/services/system-users";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import {
+  getSuccessMessageWithAttentionCheck,
+  logProcessSummaryToJobLogger,
+} from "../../../utilities";
+import { QueueConfiguration } from "@sims/sims-db";
+
+/**
+ * Retry assessments.
+ */
+@Processor(QueueNames.AssessmentWorkflowQueueRetry)
+export class AssessmentWorkflowQueueRetryScheduler extends BaseScheduler<void> {
+  constructor(
+    @InjectQueue(QueueNames.AssessmentWorkflowQueueRetry)
+    schedulerQueue: Queue<void>,
+    queueService: QueueService,
+    private readonly workflowEnqueuerService: WorkflowEnqueuerService,
+    private readonly studentAssessmentService: StudentAssessmentService,
+    private readonly systemUsersService: SystemUsersService,
+  ) {
+    super(schedulerQueue, queueService);
+  }
+
+  /**
+   * Process all assessments that were not processed in a period of time.
+   * @param job job information.
+   * @returns processing result.
+   */
+  @Process()
+  async enqueueAssessmentRetryOperations(job: Job<void>): Promise<string[]> {
+    const processSummary = new ProcessSummary();
+    try {
+      processSummary.info("Checking assessments to be queued for retry.");
+      // Check for assessments cancellations to be retried.
+      await this.executeEnqueueProcess(
+        processSummary,
+        this.workflowEnqueuerService.enqueueCancelAssessmentRetryWorkflows.bind(
+          this.workflowEnqueuerService,
+        ),
+      );
+      // Check for assessments to be started.
+      await this.executeEnqueueProcess(
+        processSummary,
+        this.workflowEnqueuerService.enqueueStartAssessmentRetryWorkflows.bind(
+          this.workflowEnqueuerService,
+        ),
+      );
+      return getSuccessMessageWithAttentionCheck(
+        "Process finalized with success.",
+        processSummary,
+      );
+    } catch (error: unknown) {
+      const errorMessage = "Unexpected error while executing the job.";
+      processSummary.error(errorMessage, error);
+      return [errorMessage];
```

Comment:
just checking shouldn't we not return, if there are any logs captured in `processSummary` with this error? or since it is happening in line 73 (logProcessSummaryToJobLogger), we don't care about it here?

Reply:
`processSummary` will be written to the POD log and Bull Dashboard logs at the below steps.
```ts
this.logger.logProcessSummary(processSummary);
await logProcessSummaryToJobLogger(processSummary, job);
```